source/as-promise/types.ts

```diff
@@ -11,6 +11,12 @@ All parsing methods supported by Got.
 */
 export type ResponseType = 'json' | 'buffer' | 'text';
 
+export interface PaginateData<R, T> {
```

Comment:
```suggestion
export interface PaginateData<ResponseType, ElementType> {
```

To make it a little bit more readable?